### ISSUE TITLE
how to implement transformer layer in oneDNN

### ISSUE THREAD

**Issue by mgrabban** [oneapi-src/oneDNN#1623]
Hello,

What would the best way to implement transformer based models using oneDNN in C/C++? Are the transformer layers already implemented (like say conv) or they need to be implemented using lower level primitives?
I am mostly interested in SYCL implementation that runs on Intel GPU (PVC) as well as NVIDIA and AMD GPUs.

Thanks.

**Comment by vpirogov** [comment]
If you are looking for a way to get oneDNN acceleration in Transformer-based model I would suggest using one of the frameworks:
* [OpenVINO(TM) toolkit](https://github.com/openvinotoolkit/openvino) for inference

* [PyTorch*](https://pytorch.org/) with [Intel Extension for PyTorch](https://github.com/intel/intel-extension-for-pytorch) for GPU training or inference.
* [Tensorflow*](https://www.tensorflow.org/) with [Intel Extension for Tensorflow](https://github.com/intel/intel-extension-for-tensorflow) for GPU training or inference.

If you are looking to write code directly in SYCL you can use the following oneDNN functions as building blocks:
* [matmul](https://oneapi-src.github.io/oneDNN/dev_guide_matmul.html)
* [layer normalization](https://oneapi-src.github.io/oneDNN/dev_guide_layer_normalization.html)
* [softmax](https://oneapi-src.github.io/oneDNN/dev_guide_softmax.html)
